_docs/developer/getting_started/vm_install_using_vagrant.md

```diff
@@ -46,212 +52,475 @@ If you using an Intel-based Mac, you will follow the instructions below._
+   
+   <br>
 
-   5. Click **Advanced Options**.
+3. Download and install the latest version of [Git](https://git-scm.com/downloads).
+   
+   <br>
+
+4. Download and install [VirtualBox 6](https://www.virtualbox.org/wiki/Download_Old_Builds_6_1)
+   and [Vagrant](https://www.vagrantup.com). Virtual Box 7 may not work
+   currently with Submitty:
+
+   > **Windows 10**
+   >
+   > * You can just go to the respective sites and download the necessary
+   > binaries.
+
+   >**MacOS**
+   >
+   > * You can either go to respective sites and download the necessary binaries
+   > or install [homebrew](http://brew.sh/) if you don't have it and then run:
+   >
+   >     ```
+   >     brew install --cask virtualbox
+   >     brew install --cask vagrant
+   >     ```
+
+   > **Ubuntu/Debian**
+   >
+   > * The Ubuntu repository does not contain the latest version of Vagrant or
+   > VirtualBox and using them may not work nor are they supported. We recommend
+   > that you either download the necessary binaries from their respective steps
+   > or follow the steps outlined below for each:
+   >
+   >    VirtualBox: <https://www.virtualbox.org/wiki/Linux_Downloads>
+   >    
+   >    Vagrant: <https://developer.hashicorp.com/vagrant/downloads> 
+   >    (if that doesn't work, try: <https://vagrant-deb.linestarve.com/>)
+   
+   <br>
+   
+5. Clone [the Submitty repository](https://github.com/Submitty/Submitty) to a
+   location on your computer (the "host").
 
-   6. Click **UEFI Firmware Settings** and restart as suggested.
+   ```
+   git clone https://github.com/Submitty/Submitty.git
+   ```
+   
+   - **NOTE:** If you are not currently part of the Submitty organization on Github, you may want to
+   [fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo)
+   the repo and use the git url from your fork instead, especially if you are looking
+   to contribute.
+
+   - _OPTIONAL: If you will be developing code in one of the [companion
+   Submitty repositories](https://github.com/Submitty) (e.g., AnalysisTools, Lichen,
+   RainbowGrades, Tutorial), also clone those repositories to the same directory._
+
+      Example Organization:
+        ```
+        home
+        └── myusername
+            └── Submitty
+                │        GIT CHECKOUT
+                ├── Submitty       (main repository)
+                ├── AnalysisTools  (optional)
+                ├── Lichen         (optional)
+                ├── RainbowGrades  (optional)
+                └── Tutorial       (optional)
+        ```
+
+      _This host directory structure will be shared / synced between
+      your host operating system and the Submitty virtual machine._
+
+   <br>
+
+6. <a name="vagrant_up_instructions"></a> Navigate into the Submitty
+repository on your computer. If on Windows, you should run CMD or
+Powershell on administrator mode.
+
+   <br>
+
+7. Decide whether your development work will *require sample
```

Comment:
I think this section could be clarified to something like:

> set the environment variable "`NO_SUBMISSIONS`" to 1 to prevent generating autograding results. 

and to better fit the flow of the instructions you could say:

> on linux/mac, you can do the following
```bash
export NO_SUBMISSIONS=1
```

> on windows, you can do
```cmd
set NO_SUBMISSIONS=1
```

and then the variable will persist in the shell session so the following command will pick it up

I also recommend moving the note on saving time to the top of the commands since people will typically copy + paste commands before reading the full text